### В данном исследовании анализируются данные о пассажирах «Титаника». Цель — определить зависимости между возрастом, классом билета, полом и вероятностью выживания, а также оценить финансовые потери компании.

In [3]:
import pandas as pd

In [4]:
df = pd.read_csv('titanic.csv') # Загружаем данные о пассажирах

In [5]:
df.drop_duplicates() # Убираем дубликаты

Unnamed: 0  survived  pclass     sex   age  sibsp  parch     fare  \
0             0         0       3    male  22.0      1      0   7.2500   
1             1         1       1  female  38.0      1      0  71.2833   
2             2         1       3  female  26.0      0      0   7.9250   
3             3         1       1  female  35.0      1      0  53.1000   
4             4         0       3    male  35.0      0      0   8.0500   
..          ...       ...     ...     ...   ...    ...    ...      ...   
886         886         0       2    male  27.0      0      0  13.0000   
887         887         1       1  female  19.0      0      0  30.0000   
888         888         0       3  female   NaN      1      2  23.4500   
889         889         1       1    male  26.0      0      0  30.0000   
890         890         0       3    male  32.0      0      0   7.7500   

    embarked   class    who  adult_male deck  embark_town alive  alone  
0          S   Third    man        True  NaN  Southampton    no  False  
1          C   First  woman       False    C    Cherbourg   yes  False  
2          S   Third  woman       False  NaN  Southampton   yes   True  
3          S   First  woman       False    C  Southampton   yes  False  
4          S   Third    man        True  NaN  Southampton    no   True  
..       ...     ...    ...         ...  ...          ...   ...    ...  
886        S  Second    man        True  NaN  Southampton    no   True  
887        S   First  woman       False    B  Southampton   yes   True  
888        S   Third  woman       False  NaN  Southampton    no  False  
889        C   First    man        True    C    Cherbourg   yes   True  
890        Q   Third    man        True  NaN   Queenstown    no   True  

[891 rows x 16 columns]

In [6]:
df.info() # Смотрим информацию о столбцах

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 16 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   Unnamed: 0   891 non-null    int64  
 1   survived     891 non-null    int64  
 2   pclass       891 non-null    int64  
 3   sex          891 non-null    object 
 4   age          714 non-null    float64
 5   sibsp        891 non-null    int64  
 6   parch        891 non-null    int64  
 7   fare         891 non-null    float64
 8   embarked     889 non-null    object 
 9   class        891 non-null    object 
 10  who          891 non-null    object 
 11  adult_male   891 non-null    bool   
 12  deck         203 non-null    object 
 13  embark_town  889 non-null    object 
 14  alive        891 non-null    object 
 15  alone        891 non-null    bool   
dtypes: bool(2), float64(2), int64(5), object(7)
memory usage: 99.3+ KB


In [7]:
df.head() # Первые 5 строк таблицы

Unnamed: 0  survived  pclass     sex   age  sibsp  parch     fare embarked  \
0           0         0       3    male  22.0      1      0   7.2500        S   
1           1         1       1  female  38.0      1      0  71.2833        C   
2           2         1       3  female  26.0      0      0   7.9250        S   
3           3         1       1  female  35.0      1      0  53.1000        S   
4           4         0       3    male  35.0      0      0   8.0500        S   

   class    who  adult_male deck  embark_town alive  alone  
0  Third    man        True  NaN  Southampton    no  False  
1  First  woman       False    C    Cherbourg   yes  False  
2  Third  woman       False  NaN  Southampton   yes   True  
3  First  woman       False    C  Southampton   yes  False  
4  Third    man        True  NaN  Southampton    no   True

## 1. Посчитайте средний возраст пассажиров сгруппированных по классам. 

In [9]:
df.groupby('class').agg({'age': 'mean'}).round(2)

age
class        
First   38.23
Second  29.88
Third   25.14

### Средний возраст пассажиров первого класса - 38, второго - 30, третьего - 25.

## 2. Выявите какого возраста пассажиры выжили больше всего.

In [12]:
df.query('survived == 1')['age'].mode()[0]

24.0

### Больше всего выжило пассажиров в возрасте 24 года. 

## 3. Выявите  среднюю плату за билеты среди всех классов.

In [15]:
df.groupby('class').agg({'fare':'mean'}).round(2)

fare
class        
First   84.15
Second  20.66
Third   13.68

### В среднем, пассажиры первого класса заплатили 84.15, второго - 20.66 и третьего - 13.68 за билет.

## 4. Выявите в каком порту было больше всего пассажиров до 40 лет.

In [18]:
df.query('age < 40').groupby('embark_town').agg({'Unnamed: 0':'count'}).reset_index().sort_values(by='Unnamed: 0', ascending = False)

embark_town  Unnamed: 0
2  Southampton         433
0    Cherbourg          95
1   Queenstown          22

###  В порту Southampton было больше всего пассажиров до 40 лет, а именно 433.

## 5. Выявите самый популярный класс для пассажиров женского пола и среднюю цену поездки по ним 

In [21]:
df.query('sex == "female"').groupby('pclass').agg({'Unnamed: 0':'count'}).reset_index().sort_values(by='Unnamed: 0', ascending = False) # Выявляем самый популярный класс среди женщин, группируя их количество по классам и сортируя по количеству по убыванию

pclass  Unnamed: 0
2       3         144
0       1          94
1       2          76

In [22]:
df.query('sex == "female"').groupby('pclass').agg({'fare':'mean'}).reset_index().sort_values(by='fare', ascending = False).round(2) # Ищем среднюю цену поездки в каждом классе среди пассажиров женского пола

pclass    fare
0       1  106.13
1       2   21.97
2       3   16.12

### Самым популярным классом среди пассажиров женского пола является третий, а средняя цена по нему составляет 16,12.

## 6. Разделив пассажиров по полам, выявите какие пассажиры имеют большую сумму платы за поездку. 

In [25]:
df.groupby('sex').agg({'fare':'sum'}).reset_index().sort_values(by='fare', ascending = False)

sex        fare
1    male  14727.2865
0  female  13966.6628

### Мужчины в сумме заплатили больше женщин почти на 800.

## 7. Теперь посчитайте сколько было заработано денег в целом.

In [28]:
profit = df['fare'].sum().round(2) ### Сумма всех заработанных денег на билетах

In [29]:
profit

28693.95

### В общем на билетах было заработано 28693.95.

## 8. Предположим, что компания решила выплатить компенсацию пострадавшим семьям. Посчитайте сколько компания должна выплатить родственникам погибших на титанике, если компания обязана выплатить 100 тыс долларов за каждого погибшего человека. 

In [32]:
deceased = df.query('survived == 0')['Unnamed: 0'].count() # Количество погибших
deceased 

549

In [33]:
compensation = deceased * 100000 # Рассчитываем компенсацию

In [34]:
compensation

54900000

### Количество погибших равно 549, соответственно компания обязана выплатить 54,900,000.

## 9. Теперь соберите вашу информацию и опишите вывод проделанного исследования. А также добавьте сумму убытка, которую можно посчитать, если вычесть общую сумму заработка и расходы на компенсации родственникам.

In [37]:
loss = compensation - profit # Отнимаем от суммы компенсации заработанные деньги

In [38]:
loss

54871306.05

### Чем старше люди, тем вероятнее они брали билеты классов повыше, соответственно и дороже. Молодые люди в основном брали билеты 3 класса. Также, люди помоложе, вероятно, имели больше шансов выжить, учитывая что больше всего выживали люди в возрасте 24 года.
### Большинство пассажиров в возрасте до 40 лет сели на порту Southampton.
### Кроме того, что третий класс был популярен среди молодых людей, можно подметить, что туда садились большинство женщин. Также, средняя цена билета такого класса у женщин была выше (16,12), чем средняя цена для обоих полов (13,67). При этом, в сумме за все билеты пассажиры-мужчины заплатили больше.
### В общем компания могла бы заработать 28693.95, но потерпела убытки в 54871306.05 в результате выплат компенсаций.